worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -656,6 +656,28 @@ class PushDataHandler extends BaseMessageHandler with Logging {
     }
   }
 
+  /**
+   * returns an array of FileWriters from partition locations along with an optional index for any FileWriter that
+   * encountered an exception.
+   */
+  private def getFileWriters(
+      partitionIdToLocations: Array[(String, PartitionLocation)])
+      : (Array[FileWriter], Option[Int]) = {
+    val fileWriters = new Array[FileWriter](partitionIdToLocations.length)
+    var i = 0
+    var exceptionFileWriterIndex: Option[Int] = None
+    while (i < partitionIdToLocations.length) {
+      val workingPartition = partitionIdToLocations(i)._2.asInstanceOf[WorkingPartition]
+      val fileWriter = workingPartition.getFileWriter
```

Comment:
prefer the following style assuming there is no performance decrease.
```suggestion
      val (_, workingPartition) = partitionIdToLocations(i)
      val fileWriter = workingPartition.asInstanceOf[WorkingPartition].getFileWriter
```